### DISCUSSION TITLE
Web socket client shuts down with RSV value errors

### DISCUSSION THREAD

**Discussion by n-srinidhi** [crossbario/autobahn-python#1640] [Q&A]
Hi! I have an autobahn web socket client and a Web socket server (Written in c#). Whilst receiving packets encoded using UTF-8 encoding, I am getting errors while decoding the payload and my web socket client shuts down.

Code:

```
def onMessage(self, payload, isBinary):
     _message=payload.decode('utf8')
```

Socket Client shuts down with following information:

`WebSocket connection closed: connection was closed uncleanly ("I dropped the WebSocket TCP connection: RSV = 2 and no extension negotiated")  WasClean False Code 1006`

Sometimes it also exits with RSV codes 3 and 7. Can anyone help with what could be the issue? I've been combing the docs but haven't made progress. 

Exception:
`ERROR:	[2024-04-16 13:27:44,178]	['Traceback (most recent call last):\n', '  File "<frozen centra_server>", line 71, in onMessage\n', "UnicodeDecodeError: 'utf-8' codec can't decode byte 0x82 in position 2087508: invalid start byte\n"]`

**Comment by oberstet** [comment]
> Can anyone help with what could be the issue?

most likely, the websocket server you are using has bugs. I would recommend checking the websocket server using

https://github.com/crossbario/autobahn-testsuite

this can be done without writing any code.

unless & until all tests return green there, it doesn't make sense to hunt for any AutobahnPython or user code issues ..

**Reply by n-srinidhi** [reply]
Hi, I'm still encountering this issue. 
1. Is it possible to pass non encoded packets  via websockets? Sometimes I am getting mixed packets from the server, So i'd rather ignore that then encounter an error and have the client shut down. 
2. Is it possible to verify size of the packet or something before i decode the payload at autobahn client end so as to avoid such errors?

Any inputs will be much appreciated!Thanks!
